### PR TITLE
New Package: Browsh

### PR DESCRIPTION
**Pull request by supechicken666**
browsh: A fully-modern text-based browser, rendering to TTY and browsers

Tested on `x86_64`

### CODE REVIEW

File: packages/browsh.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    @_arch = '386'
+  end
+    
+  @_file = "browsh_#{@_ver}_linux_#{@_arch}"
+  @_url = "https://github.com/browsh-org/browsh/releases/download/v#{@_ver}/"
+
+  source_url @_url + @_file
+  
+  def self.preflight
+    source_sha256 `curl -Ls #{@_url}/browsh_#{@_ver}_checksums.txt`.scan(/(.*)#{@_file}$/)[0][0].strip
```
Comment (uberhacker) [inline packages/browsh.rb:28]: Do we need to put this in preflight?

File: packages/browsh.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class Browsh < Package
+  description 'A fully-modern text-based browser, rendering to TTY and browsers.'
+  homepage 'https://www.brow.sh'
+  @_ver = '1.6.4'
+  version @_ver
+  license 'LGPL-2.1'
+  compatibility 'all'
+
+  depends_on 'firefox'
```
Comment (uberhacker) [inline packages/browsh.rb:11]: We need another browser to use this?

Comment (satmandu) [reply packages/browsh.rb:11]: This program is literally a text-mode proxy to a firefox instance.

Comment (uberhacker) [reply packages/browsh.rb:11]: Maybe that should be in the description?  It currently appears to be a browser.  Sounds more like a Firefox addon.

File: packages/browsh.rb (hunk 1 above)
Comment (saltedcoffii) [reply packages/browsh.rb:28]: I feel like this could just go in the normal source_sha256 spot.

File: packages/browsh.rb (hunk 2 above)
Comment (saltedcoffii) [reply packages/browsh.rb:11]: One could consider it a TUI browser that uses firefox as a backend.

File: packages/browsh.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+require 'package'
+
+class Browsh < Package
+  description 'A fully-modern text-based browser, rendering to TTY and browsers.'
+  homepage 'https://www.brow.sh'
+  @_ver = '1.6.4'
+  version @_ver
+  license 'LGPL-2.1'
+  compatibility 'all'
```
Comment (saltedcoffii) [inline packages/browsh.rb:9]: iirc `firefox` is only for x86_64 and i686 the way we have it packaged.

Comment (saltedcoffii) [reply packages/browsh.rb:9]: If anyone wants to build firefox from the source on arm machines and write that into the rb file, that would be nice for browsh and for regular firefox on arm purposes.

File: packages/browsh.rb (hunk 1 above)
Comment (supechicken) [reply packages/browsh.rb:28]: It is for speeding up actions which doesn't need to download (search, update)

Comment (uberhacker) [reply packages/browsh.rb:28]: I'm not following.  Shouldn't those operations skip it anyway?

File: packages/browsh.rb (hunk 2 above)
Comment (supechicken) [reply packages/browsh.rb:11]: I think we don't need to consider it (my purpose for adding this package is to provide a convenient way for user to install)

File: packages/browsh.rb (hunk 1 above)
Comment (supechicken) [reply packages/browsh.rb:28]: Although they will skip the download part (and checksum), the content inside the backtick will still execute by ruby

Comment (uberhacker) [reply packages/browsh.rb:28]: Why not just make it static?

File: packages/browsh.rb (hunk 3 above)
Comment (uberhacker) [reply packages/browsh.rb:9]: Building firefox from source is not a nimble project.

File: packages/browsh.rb (hunk 1 above)
Comment (uberhacker) [reply packages/browsh.rb:28]: @supechicken: I think we better make all checksums static going forward.  I see what you mean when the checksum is executed during `crew search`.  This happens on the require line in `crew` as shown below:
```
def set_package (pkgName, pkgPath)
  require pkgPath
```

Comment (supechicken) [reply packages/browsh.rb]: Change made

File: packages/browsh.rb (hunk 3 above)
Comment (uberhacker) [reply packages/browsh.rb:9]: This package will not allow arm users to install until the firefox package supports arm.  As dependencies go, so goes the parent package.  Should we just remove arm support for now?

**Pull request closed without merging**